File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PingEndpoint.java
package uk.gov.justice.digital.hmpps.oauth2server.resource.api;

import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
import org.springframework.boot.actuate.endpoint.web.annotation.RestControllerEndpoint;
import org.springframework.boot.actuate.endpoint.web.annotation.WebEndpoint;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;

import static org.springframework.http.MediaType.*;

@Configuration
@WebEndpoint(id = "ping")
public class PingEndpoint {

    @ReadOperation(produces = "plain/text")
    public String ping() {
        return "pong";
    }
}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PingEndpoint.kt
package uk.gov.justice.digital.hmpps.oauth2server.resource.api

import org.springframework.boot.actuate.endpoint.annotation.ReadOperation
import org.springframework.boot.actuate.endpoint.web.annotation.WebEndpoint
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType

@Configuration
@WebEndpoint(id = "ping")
class PingEndpoint {
  @ReadOperation(produces = [MediaType.TEXT_PLAIN_VALUE])
  fun ping(): String = "pong"
}
